delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/state/ScopedStateGet.scala

```diff
@@ -8,11 +8,22 @@ import doobie.{ConnectionIO, Get, Put}
 object ScopedStateGet {
 
   def apply[Id: Put, S: Get](tpe: EntityType, project: ProjectRef, id: Id, tag: Tag): ConnectionIO[Option[S]] =
-    sql"""SELECT value FROM scoped_states WHERE type = $tpe AND org = ${project.organization} AND project = ${project.project}  AND id = $id AND tag = $tag"""
+    sql"""SELECT value FROM scoped_states WHERE type = $tpe AND org = ${project.organization} AND project = ${project.project} AND id = $id AND tag = $tag"""
+      .query[S]
+      .option
+
+  def apply[Id: Put, S: Get](tpe: EntityType, project: ProjectRef, id: Id, rev: Int): ConnectionIO[Option[S]] =
```

Comment:
We don't store the state for every revision, we have to query for events and replay them in a Stream (the code is already in `ScopedStateStore`, it could be moved here)